### ISSUE TITLE
printf "\033c" won't wipe terminal, nor does clear nor reset

### ISSUE THREAD

**Issue by randair0x** [ish-app/ish#855]
That's pretty much the whole issue. I'd classify it as a minor display/functionality issue, but it is a practical hassle letting it run on the background & coming back to the very beginning of my interactions

**Comment by randair0x** [comment]
a more detailed explanation:
when running several commands (e.g:
```
apk add nodejs
apk add npm
apk add git
apk add nano
git clone https://github.com/example.git
cd example
npm ci
node index.js
<runtime>
^C
```
& not clearing the screen continuously, most commands & their output will stick when using `clear` (or the alternatives), messing with scrolling (when going to another app then reopening iSH, iSH will show only the topmost of the console when there's many lines in it)

**Comment by randair0x** [comment]
yes, but also clear won't clear the actual screen after running several commands. The results vary from session to session, but it looks a bit like this: [https://imgur.com/a/gcvIDhH](url)

**Comment by tbodt** [comment]
Huh, I would have expected it to not actually clear anything, and instead sort of output blank lines at the bottom to scroll past everything. There's definitely something funky going on here.

**Comment by chanil1218** [comment]
Same here. 
When I switched to another app using cmd + tab and returned back to iSH, then scroll position is not bottom where I switched but little above so that I need to scroll down to check. 
But sometimes it remains in bottom. Strange.

**Comment by tjim** [comment]
This issue is infuriating. The worst is when I switch back to iSH and I don’t realize the scroll position has changed.

Not an iOS programmer, but maybe the solution is here? https://developer.apple.com/library/archive/featuredarticles/ViewControllerPGforiPhoneOS/PreservingandRestoringState.html

I don’t see restorationIdentifier in the code, so maybe you aren’t using it.

**Comment by tbodt** [comment]
> This issue is infuriating. The worst is when I switch back to iSH and I don’t realize the scroll position has changed.
> 
> Not an iOS programmer, but maybe the solution is here? https://developer.apple.com/library/archive/featuredarticles/ViewControllerPGforiPhoneOS/PreservingandRestoringState.html
> 
> I don’t see restorationIdentifier in the code, so maybe you aren’t using it.

This is only relevant for restoring state after iSH exits, which isn't the problem here.

Pretty sure I tried to fix this once but it only inspired further hatred of webkit.

**Comment by tjim** [comment]
Maybe swap in a different terminal emulator like https://github.com/migueldeicaza/SwiftTerm

**Comment by yueyericardo** [comment]
Any updates? I’m also observing this issue
> switching to another app and switching back doesn't seem to preserve the scroll position.
